provider/github/client.go

```diff
@@ -443,3 +477,60 @@ func handleAPIError(resp *github.Response, err error, msg string) error {
 		msg,
 	)
 }
+
+// ValidateTokenPermissions checks that client has necessary permissions required by lookout
+// returns error if any is missed
+func ValidateTokenPermissions(client *Client) error {
+	// we need `repo` access to be able to read from private repositories
+	// and `repo:status` which is part of `repo` for posting statuses
+	required := []string{"repo"}
+
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
+
+	// authorizations api can be accessed only with username and password, not token
+	// read headers of any endpoint instead
+	_, r, err := client.Users.Get(ctx, "me")
+	if err != nil {
+		return err
+	}
+
+	scopes := strings.Split(r.Header.Get("X-Oauth-Scopes"), ",")
+	scopesMap := make(map[string]bool, len(scopes))
+	for _, scope := range scopes {
+		scopesMap[strings.TrimSpace(scope)] = true
+	}
+
+	for _, scope := range required {
+		_, ok := scopesMap[scope]
+		if !ok {
+			return fmt.Errorf("token doesn't have permission scope '%s'", scope)
+		}
+	}
+
+	return nil
+}
+
+// CanPostStatus check if the client has push access to a repository
+// which is required for updating status. It assumes client has correct scope.
+// returns error if it permission is missed
+func CanPostStatus(client *Client, repo *repositoryInfo) error {
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
+
+	user, err := client.Username()
+	if err != nil {
+		return err
+	}
+
+	r, _, err := client.Repositories.GetPermissionLevel(ctx, repo.Owner, repo.Name, user)
+	if err != nil {
+		return err
+	}
+
+	if r.GetPermission() != "admin" && r.GetPermission() != "write" {
+		return fmt.Errorf("client does not have write access to repository %s", repo.FullName)
```

Comment:
To be consistent with the error on `ValidateTokenPermissions` I suggest
```suggestion
		return fmt.Errorf("token doesn't have write access to repository %s", repo.FullName)
```